templates/tapestry.js

```diff
@@ -1374,329 +1384,28 @@ function tapestryTool(config){
-                                };
-    
-                                // Update the mediaButton icon to play icon
-                                updateMediaIcon(id, mediaType, 'play');
-                                seeked = true;
-    
-                                recordAnalyticsEvent('user', 'pause', 'h5p-video', id, {'time': h5pVideo.getCurrentTime()});
-                                break;
-    
-                            case h5pObj.Video.BUFFERING:
-    
-                                // Update the mediaButton icon to loading icon
-                                updateMediaIcon(id, mediaType, 'loading');
-                                break;
-                        }
-                    });
-    
-                    // Auto-play
-                    setTimeout(function(){
-                        h5pVideo.play();
-                        recordAnalyticsEvent('app', 'auto-play', 'h5p-video', id);
-                    }, 1000);
-                }
 
-            }, false);
-        } else if (mediaFormat === "embed") {
-            mediaEl = $('<iframe id="embed" src="' + mediaUrl + '" width="' + width + '" height="' + height + '" frameborder="0" allowfullscreen="allowfullscreen"><\/iframe>');
-        }
-    
-        return mediaEl;
-    }
-    
-    function createTextNodeElement(title, str) {
-        var lightboxContent = document.createElement("div");
-        var titleSection = document.createElement("div");
-        titleSection.setAttribute("id", "text-light-box-title");
-    
-        var titleText = document.createElement("h3");
-        titleText.appendChild(document.createTextNode(title));
-        titleSection.append(titleText);
-        lightboxContent.append(titleSection);
-    
-        if (str) {
-            var paragraphSection = document.createElement("div");
-            paragraphSection.setAttribute("id", "text-light-box-paragraph");
-            paragraphArray = str.split("\n\n");
-            for (var i = 0; i < paragraphArray.length; i++) {
-                var paraDiv = document.createElement("div");
-                var para = document.createElement("p");
-                para.setAttribute("id", "text-light-box-paragraph-text");
-                para.innerHTML = paragraphArray[i].replace('\n','<br>');
-                paraDiv.appendChild(para);
-                paragraphSection.appendChild(paraDiv);
-            }
-            lightboxContent.appendChild(paragraphSection);
-        }
-        return $(lightboxContent);
+    this.saveVideoProgress = function(id, currentTime, duration) {
+        updateViewedValue(id, currentTime, duration);
+        updateViewedProgress();
     }
+
+    this.recordAnalyticsEvent = recordAnalyticsEvent;
     
+    this.exitViewMode = exitViewMode;
     // Builds the view mode, including functionality to
-    function changeToViewMode(lightboxDimensions) {
+    this.changeToViewMode = function changeToViewMode(lightboxDimensions) {
```

Comment:
Same as my previous comments, please see if it's possible to move these out of here and to view instead where they fit logically.

Reply:
Same as above, changeToViewMode does some stuff with d3.